src/main/java/net/neoforged/neoforge/network/configuration/RegistryDataMapNegotiation.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.network.configuration;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import net.minecraft.core.Registry;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.network.protocol.configuration.ServerConfigurationPacketListener;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.neoforge.network.payload.KnownRegistryDataMapsPayload;
+import net.neoforged.neoforge.registries.RegistryManager;
+import net.neoforged.neoforge.registries.datamaps.DataMapType;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public record RegistryDataMapNegotiation(ServerConfigurationPacketListener listener) implements ICustomConfigurationTask {
+    public static final ResourceLocation ID = new ResourceLocation("neoforge:registry_data_map_negotiation");
+    public static final Type TYPE = new Type(ID);
+
+    @Override
+    public Type type() {
+        return TYPE;
+    }
+
+    @Override
+    public void run(Consumer<CustomPacketPayload> sender) {
+        if (listener.isVanillaConnection()) {
```

Comment:
This must check `!listener.isConnected(KnownRegistryDataMapsPayload.ID)` instead of `listener.isVanillaConnection()` as older NF clients need to be treated the same way as vanilla clients since they don't have this system in place and as such won't be able to respond to the packet in order to mark the task completed